README.md

```diff
@@ -15,6 +48,568 @@ go get -u github.com/src-d/kallax/...
+
+### Querying JSON
+
+You can query arbitrary JSON using the JSON operators defined in the [kallax](https://godoc.org/github.com/src-d/go-kallax) package. The schema of the JSON (if it's a struct, obviously for maps it is not) is also generated.
+
+```go
+q := NewPostQuery().Where(kallax.JSONContainsAnyKey(
+        Schema.Post.Metadata,
+        "foo", "bar",
+))
+```
+
+## Transactions
+
+To execute things in a transaction the `Transaction` method of the model store can be used. All the operations done using the store provided to the callback will be run in a transaction.
+If the callback returns an error, the transaction will be rolled back.
+
+```go
+store.Transaction(func(s *UserStore) error {
+        if err := s.Insert(user1); err != nil {
+                return err
+        }
+
+        return s.Insert(user2)
+})
+```
+
+The fact that a transaction receives a store with the type of the model can be a problem if you want to store several models of different types. You can, indeed, create new stores of the other types, but do so with care. Do not use the internal `*kallax.Store`, as it does not perform any type checks or some of the operations the concrete type stores do.
+
+```go
+store.Transaction(func(s *UserStore) error {
+        postStore := &PostStore{s.Store}
+
+        for _, p := range posts {
+                if err := postStore.Insert(p); err != nil {
+                        return err
+                }
+        }
+
+        return s.Insert(user)
+})
+```
+
+`Transaction` can be used inside a transaction, but it does not open a new one, reuses the existing one.
+
+## Caveats
+
+* It is not possible to use slices or arrays of types that are not one of these types:
+  * Basic types (e.g. `[]string`, `[]int64`) (except for `rune`, `complex64` and `complex128`)
+  * Types that implement `sql.Scanner` and `driver.Valuer`
+  The reason why this is not possible is because kallax implements support for arrays of all basic Go types by hand and also for types implementing `sql.Scanner` and `driver.Valuer` (using reflection in this case), but without having a common interface to operate on them, arbitrary types can not be supported.
+  For example, consider the following type `type Foo string`, using `[]Foo` would not be supported. Know that this will fail during the scanning of rows and not in code-generation time for now. In the future, might be moved to a warning or an error during code generation.
+  Aliases of slice types are supported, though. If we have `type Strings []string`, using `Strings` would be supported, as a cast like this `([]string)(&slice)` it's supported and `[]string` is supported.
+* `time.Time` and `url.URL` need to be used as is. That is, you can not use a type `Foo` being `type Foo time.Time`. `time.Time` and `url.URL` are types that are treated in a special way, if you do that, it would be the same as saying `type Foo struct { ... }` and kallax would no longer be able to identify the correct type.
+
```

Comment:
Arrays of arrays, and slices of lices (and any combination of these) are not supported by the db driver, so it can not be handled by Kallax.